chainerx_cc/chainerx/routines/logic.cc

```diff
@@ -25,11 +25,19 @@ class LogicBinaryImpl {
     }
 };
 
+void CheckLogicDtypes(DtypeKind kind1, DtypeKind kind2) {
+    if ((kind1 == DtypeKind::kBool) != (kind2 == DtypeKind::kBool)) {
+        throw DtypeError{"Comparison does not support Boolean and another mixed dtypes"};
```

Comment:
This sentence does not sound natural. How about this:
"Comparison of bool and non-bool dtypes is not supported."

Reply:
Fixed.